src/Macro.php

```diff
@@ -24,14 +24,39 @@ public function __construct(
         $interfaces = array()
     ) {
         parent::__construct($method, $alias, $class, $methodName, $interfaces);
+
+        // Remove the static tag
+        $this->phpdoc->deleteTag(
+            $this->phpdoc->getTagsByName('static')[0]
+        );
     }
 
     /**
      * @param \ReflectionFunctionAbstract $method
      */
     protected function initPhpDoc($method)
     {
-        $this->phpdoc = new DocBlock($method);
+        // Init DocBlock only with a description
+        $this->phpdoc = new DocBlock('/** No description available. */');
+
+        // Add macro parameters
+        foreach ($method->getParameters() as $parameter) {
+            $type = $parameter->hasType() ? $parameter->getType() : 'mixed';
```

Comment:
```
./artisan ide-helper:generate

   ErrorException

  Function ReflectionType::__toString() is deprecated

  at vendor/barryvdh/laravel-ide-helper/src/Macro.php:45
    41|
    42|         // Add macro parameters
    43|         foreach ($method->getParameters() as $parameter) {
    44|             $type = $parameter->hasType() ? $parameter->getType() : 'mixed';
  > 45|             $type .= $parameter->hasType() && $parameter->getType()->allowsNull() ? '|null' : '';
    46|
    47|             $name = $parameter->isVariadic() ? '...' : '';
    48|             $name .= '$' . $parameter->getName();
    49|

      +21 vendor frames
  22  artisan:37
      Illuminate\Foundation\Console\Kernel::handle()
```

I guess this is dependent on the PHP version? It's "just" a deprecation but people might run with E_ALL and thus trip over this.

This fixes it:
```suggestion
            $type = $parameter->hasType() ? $parameter->getType()->getName() : 'mixed';
```

Reply:
I have changed this to your suggestion.